pkg/skaffold/jib/jib.go

```diff
@@ -229,3 +231,15 @@ func relativize(path string, roots ...string) (string, error) {
 	}
 	return "", errors.New("could not relativize path")
 }
+
+// isOnInsecureRegistry checks if the given image specifies an insecure registry
+func isOnInsecureRegistry(image string, insecureRegistries map[string]bool) bool {
+	ref, err := name.ParseReference(image)
+	if err != nil {
+		// ignore the error as the image should have been validated before this
```

Comment:
I wonder if we can just crash, as assumption/assertion violation.

Reply:
@balopat should we `panic()`?

Reply:
I don't think we should ever `panic()` on purpose in skaffold, but we should probably exit the build here since we know it won't succeed. I think the solutions here are:
* change this to return `(bool, error)`, check the error in the caller and conditionally exit (less work, but doesn't tell the whole story)
* do some proper image validation before we ever get this far in the builds (more work, but the "right" way)

up to you which one you want to do.

Reply:
`(bool,error)` it is!  I'm pretty sure the validation has been done long before this.